test/Data/Ord/CompatSpec.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE CPP #-}
+module Data.Ord.CompatSpec (main, spec) where
+
+import           Test.Hspec
+
+#if MIN_VERSION_base(4,6,0)
+import           Data.Ord.Compat
+#endif
+
+main :: IO ()
+main = hspec spec
+
+spec :: Spec
+#if MIN_VERSION_base(4,6,0)
```

Comment:
I am too, although I think I'll wait and do that in a separate pull request, since there'd be quite a few data types I'd want to redefine (off the top of my head, `Bifunctor`, `Identity`, `Alt`, `Void`, `Nat`, etc.).